//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated from a template.
//
//     Manual changes to this file may cause unexpected behavior in your application.
//     Manual changes to this file will be overwritten if the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

namespace Contacts.Data
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.ComponentModel.DataAnnotations;
    using System.Data.Entity;
    using System.Linq;

    public partial class Customer
    {
        public int id { get; set; }
        public string NameFirst { get; set; }
        public string NameMiddle { get; set; }
        public string NameLast { get; set; }
        public string Phone { get; set; }
        public Nullable<int> CreditScore { get; set; }
        public Nullable<System.DateTime> Stamp { get; set; }

        public static List<Customer> GetAll()
        {
            using (WpfEntities db = new WpfEntities())
            {
                return db.Customers.OrderByDescending(Row => Row.id).ToList();
            }
        }

        public static dynamic GetByID(int id)
        {
            WpfEntities db = new WpfEntities();

            Customer Customer = db.Customers.Find(id);

            db.Dispose();

            if (Customer == null) return null;

            return Customer;
        }

        public dynamic Add()
        {
            if (this.id > 0) return false;

            WpfEntities db = new WpfEntities();

             try
            {
                db.Customers.Add(this);
                db.SaveChanges();

                return this.id;
            }
            catch (Exception Exception)
            {
                //JDR: Exception can be logged here
                Console.WriteLine("***EROR_CATCH*** - An error occurred: '{0}'", Exception);

                return false;
            }
            finally
            {
                db.Dispose();
            }
        }

        public bool Save()
        {
            WpfEntities db = new WpfEntities();

            try
            {
                db.Entry(this).State = EntityState.Modified;

                db.SaveChanges();

                return true;
            }
            catch (Exception Exception)
            {
                //JDR: Exception can be logged here
                Console.WriteLine("***EROR_CATCH*** - An error occurred: '{0}'", Exception);

                return false;
            }
            finally
            {
                db.Dispose();
            }
        }

        public bool Delete()
        {
            WpfEntities db = new WpfEntities();

            try
            {
                db.Entry(this).State = EntityState.Deleted;

                db.SaveChanges();

                return true;
            }
            catch (Exception Exception)
            {
                //JDR: Exception can be logged here
                Console.WriteLine("***EROR_CATCH*** - An error occurred: '{0}'", Exception);

                return false;
            }
            finally
            {
                db.Dispose();
            }
        }
    }
}
